src/Parser/MarksmanshipHunter/Modules/FocusChart/MaxFocus.js

```diff
@@ -0,0 +1,11 @@
+import Module from 'Parser/Core/Module';
+
+class MaxFocus extends Module {
+  on_byPlayer_cast(event){
+    if (event.sourceID === this.owner.player.id && (event.type === 'cast' || event.type === 'energize')){
+      this._focus = event.classResources[0]['max'];
```

Comment:
I recommend looping over class resources to prevent this breaking for whatever reason in the future. A simple boss mechanic that adds a second class resource, some item, legendary set bonus or anything that messes with class resources could break this.

```jsx
event.classResources.forEach(classResource => {
  if (classResource.type === RESOURCE_TYPES.FOCUS) {
    if (classResource.max > this.max) {
      this.max = classResource.max;
    }
  }
});
```